src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -98,44 +99,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
     String url = videoUrl.get();
 
-    Query query = new Query(Lecture.ENTITY_KIND);
+    Query query = new Query(LectureUtil.KIND);
     PreparedQuery results = datastore.prepare(query);
     Iterable<Entity> resultsIterable = results.asIterable();
 
     for (Entity lecture : resultsIterable) {
-      if (lecture.getProperty(Lecture.PROP_VIDEO_URL).equals(url)) {
+      if (lecture.getProperty(LectureUtil.VIDEO_URL).equals(url)) {
         return Optional.of(lecture);
       }
     }
     return Optional.empty();
   }
 
-  @VisibleForTesting
-  /** Creates and returns {@code lectureEntity} using parameters found in {@code request}. */
+  /** Returns {@code lectureEntity} using parameters found in {@code request}. */
   // TODO: Send errors as a response if fields are empty.
-  // TODO: Move to LectureEntityCreator class.
-  protected Entity createLectureEntity(HttpServletRequest request) {
+  private Entity getLectureEntityFromRequest(HttpServletRequest request) {
     Optional<String> optionalLectureName = getParameter(request, NAME_INPUT);
     String lectureName = optionalLectureName.isPresent() ? optionalLectureName.get() : "";
     Optional<String> optionalVideoUrl = getParameter(request, LINK_INPUT);
     String videoUrl = optionalVideoUrl.isPresent() ? optionalVideoUrl.get() : "";
     Optional<String> optionalVideoId = getVideoId(videoUrl);
     String videoId = optionalVideoId.isPresent() ? optionalVideoId.get() : "";
-
-    Entity lectureEntity = new Entity(Lecture.ENTITY_KIND);
-    lectureEntity.setProperty(Lecture.PROP_NAME, lectureName);
-    lectureEntity.setProperty(Lecture.PROP_VIDEO_URL, videoUrl);
-    lectureEntity.setProperty(Lecture.PROP_VIDEO_ID, videoId);
-    return lectureEntity;
+    return LectureUtil.createEntity(lectureName, videoUrl, videoId);
```

Comment:
I would have this method take in as input Optional therefore if other methods also obtain an Optional field they do not need to convert it to Strings. This reduces code duplication.

Reply:
Jay let me know there is a TODO to return an error if the fields are empty.